### PR TITLE
Improve JSDocs to be more descriptive and better overall

### PR DESCRIPTION
**Pull request by bitwise74**
I've improved all relevant JSDocs across files `Player.ts`, `Poru.ts`, `Queue.ts`, and `Response.ts` so they have a more descriptive JSDoc comment. I also fixed some spelling issues/sentences that made no sense.

## Breaking changes:
- I've changed the `player.stop` method to `player.skip` instead as `stop` implies a complete stop of the player's playback where `skip` implies skipping the currently playing song
- I've removed the unnecessary parameter from `poru.init` (that being the client parameter) as it's already provided in poru's constructor

### CODE REVIEW

File: src/Player/Player.ts [hunk 1]
```diff
@@ -1,560 +1,576 @@
...
+import { Poru, ResolveOptions, EventData, ConnectionOptions } from "../Poru"
+import { Node } from "../Node/Node"
+import { Track } from "../guild/Track"
+import { Connection } from "./Connection"
+import Queue from "../guild/Queue"
+import { EventEmitter } from "events"
+import { Filters } from "./Filters"
+import { Response, LoadTrackResponse } from "../guild/Response"
+
+type Loop = "NONE" | "TRACK" | "QUEUE"
```
Comment (UnschooledGamer) [inline src/Player/Player.ts:10]: Why was the jsdoc removed of `loop` type?

File: src/Poru.ts [hunk 2]
```diff
@@ -516,10 +479,10 @@ export class Poru extends EventEmitter {
     }
 
     /**
-     * Resolve a track from poru instance
-     * @param {ResolveOptions} param0  ResolveOptions
-     * @param {Node | undefined} node Node or undefined
-     * @returns {Promise<Response>} The Response of the resolved tracks
+     * Resolves a track.
+     * @param {ResolveOptions} options - Options for resolving tracks.
```
Comment (UnschooledGamer) [inline src/Poru.ts:483]: This is probably an invalid param as there no param called as `options` defined in the `resolve` function

File: src/guild/Queue.ts [hunk 3]
```diff
@@ -1,58 +1,53 @@
 import { Track } from "./Track"
 
 export default class Queue extends Array<Track> {
-    constructor() {
-        super(...arguments);
-    }
-
```
Comment (UnschooledGamer) [inline src/guild/Queue.ts:7]: can I know why were these lines removed?

Comment (UnschooledGamer) [reply src/guild/Queue.ts:7]: Sorry, Just came to know that JS by default does that.

**Comment by UnschooledGamer** [comment]
> I've improved all relevant JSDocs across files `Player.ts`, `Poru.ts`, `Queue.ts`, and `Response.ts` so they have a more descriptive JSDoc comment. I also fixed some spelling issues/sentences that made no sense.
> 
> ## Breaking changes:
> * I've changed the `player.stop` method to `player.skip` instead as `stop` implies a complete stop of the player's playback where `skip` implies skipping the currently playing song
> * I've removed the unnecessary parameter from `poru.init` (that being the client parameter) as it's already provided in poru's constructor

Overall looks Good, Discussing about the breaking changes firstly the `player.stop` 
> "implies a complete stop of the player's playback"

true, that's actually what it does completely stops the current playing track. 
secondly `client` parameter is present probably to provide the updated `client` data as `<Poru>.init` would be run once the discord client is ready(fully connected to discord) and before that `client` won't have access to it's `user id`.

I might be wrong don't mind it and correct me.

Apart from this I'll wait 24h for @parasop to review if he doesn't in 24h I'll merge this. (If I don't or forgot to do so Kindly Reminder me 😅)

-- Thanks UnschooledGamer

**Comment by bitwise74** [comment]
> secondly client parameter is present probably to provide the updated client data as <Poru>.init would be run once the discord client is ready(fully connected to discord) and before that client won't have access to it's user id

The thing about the client (at least for discord.js, not sure about other libraries) is that you can't access the `user` property if it's not initialized. I didn't see any other references to the `client` parameter inside of the `init` method which made me think it's superfluous.

**Comment by bitwise74** [comment]
> true, that's actually what it does completely stops the current playing track.

As a user of the library I avoided the `stop` method as I figured it would completely stop the player and disconnect it from the voice channel similar to what `destroy` does hence why I renamed it to a more straight-forward `skip` instead. It may just be me but I feel like `skip` would work better here.

**Comment by UnschooledGamer** [comment]
> > secondly client parameter is present probably to provide the updated client data as <Poru>.init would be run once the discord client is ready(fully connected to discord) and before that client won't have access to it's user id
> 
> The thing about the client (at least for discord.js, not sure about other libraries) is that you can't access the `user` property if it's not initialized. I didn't see any other references to the `client` parameter inside of the `init` method which made me think it's superfluous. 

It's most likely the same as after the bot successfully connects discord Gateway it returns the information about the user id etc.

**Comment by bitwise74** [comment]
Probably yeah. So the client parameter seems to be redundant here then.

**Comment by UnschooledGamer** [comment]
> > true, that's actually what it does completely stops the current playing track.
> 
> As a user of the library I avoided the `stop` method as I figured it would completely stop the player and disconnect it from the voice channel similar to what `destroy` does hence why I renamed it to a more straight-forward `skip` instead. It may just be me but I feel like `skip` would work better here.

`stop` just tells Lavalink to stop the current playing song that's all. The disconnection afterwards is probably by code written somewhere.

**Comment by bitwise74** [comment]
> > > true, that's actually what it does completely stops the current playing track.
> > 
> > 
> > As a user of the library I avoided the `stop` method as I figured it would completely stop the player and disconnect it from the voice channel similar to what `destroy` does hence why I renamed it to a more straight-forward `skip` instead. It may just be me but I feel like `skip` would work better here.
> 
> `stop` just tells Lavalink to stop the current playing song that's all. The disconnection afterwards is probably by code written somewhere.

I'm not saying `stop` disconnects the bot from the voice channel, just saying that's what it *feels like* it would do

**Comment by UnschooledGamer** [comment]
> 
> > > > true, that's actually what it does completely stops the current playing track.
> > > 
> > > 
> > > As a user of the library I avoided the `stop` method as I figured it would completely stop the player and disconnect it from the voice channel similar to what `destroy` does hence why I renamed it to a more straight-forward `skip` instead. It may just be me but I feel like `skip` would work better here.
> > 
> > `stop` just tells Lavalink to stop the current playing song that's all. The disconnection afterwards is probably by code written somewhere.
> 
> I'm not saying `stop` disconnects the bot from the voice channel, just saying that's what it *feels like* it would do
> 
> 

Ah, doesn't feel for me. stop means it stops the player and destroy means it destroys the player.

**Comment by bitwise74** [comment]
Still, I feel like `skip` is just a better name for a method if it's supposed to skip the currently playing song and play the next one in the queue like normal
